src/services/userTransform.ts

```diff
@@ -73,6 +74,13 @@ export class UserTransformService {
           messageIds,
         };
 
+        const generateNewMetadata = (response: UserTransformationResponse) => ({
+          ...eventsToProcess[0]?.metadata,
+          messageId: generateUUID(),
+          eventName: response.transformedEvent.event,
+          eventType: response.transformedEvent.type,
+        });
+
```

Comment:
Looks like, we are already counting the newly added event as successful with empty event name. The metadata [is present with all fields set as undefined](https://github.com/rudderlabs/rudder-transformer/blob/develop/src/util/ivmFactory.js#L101). 

Generating new messageId change might not be required ? The benefit of this change is that we might be able to track the metrics against correct event name ?